docs-vim.js

```diff
@@ -57,36 +71,91 @@ vim.normal_keydown = function (e) {
 	vim.num = "";
     }
 
-    if (e.key == "V" && e.shift) {
-        console.log("select");
+    return false;
+};
+
+// Called in visual mode.
+vim.visual_keydown = function (e) {
+    if (e.key.match(/F\d+/)) {
+        // Pass through any function keys.
+        return true;
+    }
+
+    if (e.key == "Escape") {
+	// Escape visual mode.
+        vim.switchToNormalMode();
+    }
+
+    vim.currentSequence += e.key;
+    if (vim.currentSequence == vim.keys.escapeSequence) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        vim.switchToNormalMode();
+        return false;
+    }
+    if (vim.keys.escapeSequence.indexOf(vim.currentSequence) != 0) {
+        vim.currentSequence = e.key;
+    }
+
+    e.preventDefault();
+    e.stopPropagation();
+
+    var keyMap = { "Backspace": "ArrowLeft", "x": "Delete" };
+    keyMap[vim.keys.move[0]] = "ArrowLeft";
+    keyMap[vim.keys.move[1]] = "ArrowDown";
+    keyMap[vim.keys.move[2]] = "ArrowUp";
+    keyMap[vim.keys.move[3]] = "ArrowRight";
+
+    if (e.key.match(/\d+/)) {
+        vim.num += e.key.toString();
+    }
+
+    if (e.key in keyMap) {
+        e.key = keyMap[e.key];
+    }
+
+    if (e.key.indexOf("Arrow") == 0 || e.key == "Delete") {
+	if (vim.num.length == 0 || isNaN(vim.num)) {
+            vim.num = "1";
+	}
+	for (var i = 0; i < Number(vim.num); i++) {
+	    if (e.key.indexOf("Arrow") == 0) {
+                docs.pressKey(docs.codeFromKey(e.key), false, true);
+	    } else {
+                docs.pressKey(docs.codeFromKey(e.key));
+		// Switch to normal mode when 'x' pressed
+		vim.switchToNormalMode();
+	    }
+	}
+	vim.num = "";
     }
 
     return false;
```

Comment:
There seem to be a bunch of `\t` tabs instead of spaces, can we switch them over for consistency?